src/eventSchemas/models/schemaCodeGenUtils.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// TODO: This is fragile. Very fragile. But it is necessary to get Schemas service launched, and we've evaluated all other tradeoffs
+// This will be done on the server-side as soon as we can, but for now the client needs to do this
+export class SchemaCodeGenUtils {
+    private readonly SCHEMA_PACKAGE_PREFIX = 'schema'
+    private readonly AWS = 'aws'
+    private readonly PARTNER = 'partner'
+    private readonly AWS_PARTNER_PREFIX = `${this.AWS}.${this.PARTNER}-` // dash suffix because of 3p partner registry name format
+    private readonly AWS_EVENTS_PREFIX = `${this.AWS}.` // . suffix because of 1p event registry schema format
+
+    public buildSchemaPackageName(schemaName: string) {
+        const builder = new CodeGenPackageBuilder()
+        builder.append(this.SCHEMA_PACKAGE_PREFIX)
+        this.buildPackageName(builder, schemaName)
+
+        return builder.getBuilder()
+    }
+
+    private buildPackageName(builder: CodeGenPackageBuilder, schemaName: string) {
+        if (this.isAwsPartnerEvent(schemaName)) {
+            this.buildPartnerEventPackageName(builder, schemaName)
+        } else if (this.isAwsEvent(schemaName)) {
+            this.buildAwsEventPackageName(builder, schemaName)
+        } else {
+            this.buildCustomPackageName(builder, schemaName)
+        }
+    }
+
+    private isAwsPartnerEvent(schemaName: string): Boolean {
+        return schemaName.startsWith(this.AWS_PARTNER_PREFIX)
+    }
+
+    private buildPartnerEventPackageName(builder: CodeGenPackageBuilder, schemaName: string) {
+        const partnerSchemaString = schemaName.substring(this.AWS_PARTNER_PREFIX.length)
+
+        builder
+            .append(this.AWS)
+            .append(this.PARTNER)
+            .append(partnerSchemaString)
+    }
+
+    private isAwsEvent(name: string): Boolean {
+        return name.startsWith(this.AWS_EVENTS_PREFIX)
+    }
+
+    private buildAwsEventPackageName(builder: CodeGenPackageBuilder, schemaName: string) {
+        const awsEventSchemaParts = schemaName.split('.')
+        for (const part of awsEventSchemaParts) {
+            builder.append(part)
+        }
+    }
+
+    private buildCustomPackageName(builder: CodeGenPackageBuilder, schemaName: string) {
+        builder.append(schemaName)
+    }
+}
+
+class CodeGenPackageBuilder {
+    private builder = ''
+    private readonly formatter = new IdentifierFormatter()
+
+    public getBuilder() {
+        return this.builder
+    }
+
+    public append(segment: String): CodeGenPackageBuilder {
```

Comment:
In theory this would only happen if the partner event schema or the customer uploaded schema has an empty name, which they shouldn't be able to create in the first place.